zircon.core/src/commonMain/kotlin/org/hexworks/zircon/api/builder/fragment/ColorThemeSelectorBuilder.kt

```diff
@@ -0,0 +1,52 @@
+package org.hexworks.zircon.api.builder.fragment
+
+import org.hexworks.zircon.api.Fragments
+import org.hexworks.zircon.api.behavior.Themeable
+import org.hexworks.zircon.api.builder.Builder
+import org.hexworks.zircon.api.data.Position
+import org.hexworks.zircon.api.fragment.MultiSelect
+import org.hexworks.zircon.api.fragment.builder.FragmentBuilder
+import org.hexworks.zircon.internal.resource.ColorThemeResource
+import kotlin.jvm.JvmStatic
+
+// constructor could be private but java can not handle private vararg constructors
+class ColorThemeSelectorBuilder(
+        width: Int,
+        vararg componentsToUpdate: Themeable
+) : FragmentBuilder<MultiSelect<ColorThemeResource>, ColorThemeSelectorBuilder> {
+
+    companion object {
+        @JvmStatic
+        fun newBuilder(width: Int, vararg componentsToUpdate: Themeable) = ColorThemeSelectorBuilder(width, *componentsToUpdate)
+    }
+
+    private val multiSelectBuilder: MultiSelectBuilder<ColorThemeResource>
+
+    init {
+        require(componentsToUpdate.isNotEmpty()) {
```

Comment:
I think this should not be mandatory. There are scenarios when the user doesn't have an initial group of components.